src/timing.h

```diff
@@ -318,16 +320,16 @@ STATIC_INLINE void _jl_timing_counts_ctor(jl_timing_counts_t *block) JL_NOTSAFEP
 #endif
 }
 
-STATIC_INLINE void _jl_timing_counts_destroy(jl_timing_counts_t *block, int subsystem) JL_NOTSAFEPOINT {
-    jl_timing_counts[subsystem] += block->total;
+STATIC_INLINE void _jl_timing_counts_destroy(jl_timing_counts_t *block, int event, uint64_t t) JL_NOTSAFEPOINT {
+    jl_atomic_fetch_add_relaxed(jl_timing_self_counts + event, block->total);
+    jl_atomic_fetch_add_relaxed(jl_timing_full_counts + event, t - block->t0);
```

Comment:
`block->t0` might be uninitialized if the event was created but not started, but this function is supposed to do nothing in that case

The smoking gun I noticed was this:
```
GC_IncrementalSweep      ,             24332148 ( 0.23 %),      158290379112508 (1464937.27 %)
```